source/manuals/programming-languages/html.html.md.erb

```diff
@@ -1,85 +1,263 @@
+
+```html
+<picture>
+  <source srcset="example-320w.webp 320w,
+                  example-480w.webp 480w,
+                  example-800w.webp 800w"
+    sizes="(max-width: 320px) 280px,
+           (max-width: 480px) 440px,
+           800px"
+    type="image/webp" />
+  <!-- Fallback for browsers that support srcset but not webp -->
+  <source srcset="example-320w.jpg 320w,
+                  example-480w.jpg 480w,
+                  example-800w.jpg 800w"
+    sizes="(max-width: 320px) 280px,
+           (max-width: 480px) 440px,
+           800px"
+    type="image/jpeg" />
+  <!-- Fallback for browsers that do not support srcset -->
+  <img src="fallback.jpg" alt="This is my alt text" />
+</picture>
+```
+
+The `alt` tag should be used to describe the image to visually impaired users. If we were doing
+a presentation with a slide of the image but had a blind person in the room, how would you make
+sure that person wasn't missing any of the information?
+
+It is perfectly acceptable to have an empty `alt=""` attribute if the image is a background or
+presentational image. This explicitly tells the screen reader that there is no alternative text
+and the image gets skipped.
+
+Note that an empty `alt=""` is _not_ the same as having no alt attribute. If the attribute is
+missing, screen readers will read out the image filename instead, which is often not very useful.
+
+### Buttons
+
+Buttons should be used for client-side interactivity that doesn't change the destination of the
+page (e.g. a close button on a modal).
+
+### Links
+
+Links should be used for navigating the user to a new page (hyperlinks), or to a different target
+on the same page (anchor links). `<a href="#">` should be avoided!
+
+Links should never open new tabs or windows. The user therefore retains control of their experience.
+
+Links to third-party websites should have an appropriate icon and off-screen text so that users
+know before they click.
+
+### Visually hidden elements
+
+Sometimes it can be helpful to provide additional content for screen reader users
+that might not be needed for sighted users. For example, the 'Skip to main content'
+link covered earlier.
+
+Do **not** use `display: none` on this text, as this will get ignored by screen readers.
+
+Instead, use CSS which ensures the text gets 'conceptually' rendered, even if the result
+is not visible on the screen. You can use the [Sass mixins from govuk-frontend](https://github.com/alphagov/govuk-frontend/blob/master/package/helpers/_visually-hidden.scss)
+to ensure that you're using the latest best practice CSS for visually hidden text, or copy
+the CSS below.
 
-It's semantically more accurate to use `<strong>` than `<em>` and this usage is a reasonably common convention.
+```css
+position: absolute;
+width: 1px;
+height: 1px;
+margin: 0;
+padding: 0;
+overflow: hidden;
+clip: rect(0 0 0 0);
+clip-path: inset(50%);
+border: 0;
+white-space: nowrap;
+```
```

Comment:
You're already linking to the code above, might as well remove this code example - it'll save keeping it up to date if it changes.

Reply:
Done.